network.go

```diff
@@ -832,35 +832,84 @@ func (n *network) updateSvcRecord(ep *endpoint, localEps []*endpoint, isAdd bool
 	}
 
 	c := n.getController()
-	sr, ok := c.svcDb[n.ID()]
-	if !ok {
+	c.Lock()
```

Comment:
Well it reads better. `updateSvcRecord` works at the endpoint level and takes care of both add and delete and then once that is decided you pushing it into network state. Also there is really no reason to get controller instance and init svcDB in `updateSvcRecord`